src/single.c

```diff
@@ -538,6 +543,16 @@ static int single_process_buffer(struct db_salt *salt)
 	struct db_keys *keys;
 	size_t size;
 
+	if (++recurse_depth > max_recursion && retest_guessed) {
+		log_event("- Disabled SingleRetestGuessed due to deep recursion");
+		if (john_main_process)
+			fprintf(stderr,
+"Warning: Disabled SingleRetestGuessed due to deep recursion. You can run\n"
+"         '--loopback --rules=none' later on instead.\n");
+
```

Comment:
NOTE, we should add a static module value, so that we only issue the log_event() 1 time, and the fprintf(stderr...) 1 time.

The recurse depth grows, and if there are more than 10000 hashes in the run, then when 'most' buffers are almost full, and a password is cracked, then it will 'trigger' a cascade recursive drop (i.e., in recursion, and each thing being recursed cracks something).  So this recurse depth will be hit, and the retesting stopped.  Then things clear up.  Then a short while later (could be instant), recursion level again caps out, and the error is listed.  So this will end up being almost a DOS, where the log will grow incredibly fast, and the stderr will get printf's incredibly fast.